src/api/layer.js

```diff
@@ -376,12 +376,21 @@ export default class Layer {
     }
 
     _addToMGLMap(map, beforeLayerID) {
-        if (map.isStyleLoaded()) {
+        this._isSourceLoaded = false;
+        
+        map.on('sourcedata', this._onMapSourcedata);
+        map.on('load', this._onMapLoad.bind(this, map, beforeLayerID));
+    }
+
+    _onMapSourcedata(event) {
+        this._isSourceLoaded = event.isSourceLoaded;   
+    }
+
+    _onMapLoad(map, beforeLayerID) {
+        if (map.isStyleLoaded() || this._isSourceLoaded) {
```

Comment:
I think ` this._isSourceLoaded` is always `undefined` because [this](https://github.com/CartoDB/carto-vl/pull/623/files#r199458446) (pun intended) :trollface: